source/common/formatter/substitution_formatter.cc

```diff
@@ -1720,16 +1753,31 @@ GrpcStatusFormatter::formatValue(const Http::RequestHeaderMap&,
   if (!grpc_status.has_value()) {
     return unspecifiedValue();
   }
-  if (format_as_number_) {
+
+  switch (format_) {
+  case CamelString: {
+    const auto grpc_status_message = Grpc::Utility::grpcStatusToString(grpc_status.value());
+    if (grpc_status_message == EMPTY_STRING || grpc_status_message == "InvalidCode") {
+      return ValueUtil::stringValue(std::to_string(grpc_status.value()));
+    }
+    return ValueUtil::stringValue(grpc_status_message);
+  }
+  case SnakeString: {
+    const auto grpc_status_message =
+        absl::StatusCodeToString(static_cast<absl::StatusCode>(grpc_status.value()));
+    if (grpc_status_message == EMPTY_STRING) {
+      return ValueUtil::stringValue(std::to_string(grpc_status.value()));
```

Comment:
Is there a test case which takes this path (when grpc_status_message is EMPTY_STRING)?

Reply:
Yes, see https://github.com/envoyproxy/envoy/pull/22333/files#diff-acf64a22c70ac33d2a2484a9904fd67cd854dcb68561834756af637850389dd4R2448